daemon/cmd/daemon_main.go

```diff
@@ -1611,13 +1611,14 @@ var daemonCell = cell.Module(
 
 	cell.Provide(
 		newDaemonPromise,
-		newRestorerPromise,
+		promise.New[endpointstate.Restorer],
 		func() k8s.CacheStatus { return make(k8s.CacheStatus) },
 		newSyncHostIPs,
 	),
 	// Provide a read-only copy of the current daemon settings to be consumed
 	// by the debuginfo API
 	cell.ProvidePrivate(daemonSettings),
+	cell.Invoke(registerEndpointStateResolver),
```

Comment:
All of this (including the `Provide` above) can be achieved via:

```suggestion
	cell.Provide(func(p promise.Promise[*Daemon]) promise.Promise[endpointstate.Restorer] {
		return promise.Map(p, func(d *Daemon) endpointstate.Restorer {
			return d
		})
	}),
```

Reply:
Tried this, caused `go test ./...` hang on `daemon/cmd`.